### PR TITLE
Please review implementation for Marker events

### PR DESCRIPTION
**Pull request by akurlykov**
Please review the proposed implementation for handling marker events.
Here I opted out from proposed handling where Marker is retrieved with @Marker("name") annotation, instead markers can be accessed from DecisionContext. Later this can be extended to get all the markers and their details.

### CODE REVIEW

File: swiffer-api/src/test/java/com/solambda/swiffer/api/internal/decisions/EventHandlerTypeFactoryTest.java [hunk 1]
```diff
@@ -11,18 +11,23 @@
...
 import com.solambda.swiffer.api.internal.context.identifier.ActivityName;
+import com.solambda.swiffer.api.internal.context.identifier.MarkerName;
 import com.solambda.swiffer.api.internal.context.identifier.SignalName;
 import com.solambda.swiffer.api.internal.context.identifier.TimerName;
 import com.solambda.swiffer.api.internal.context.identifier.WorkflowName;
 
 public class EventHandlerTypeFactoryTest {
 	private static final String SIGNAL1 = "signal1";
 	private static final String TIMER1 = "timer1";
+	private static final String MARKER1 = "timer1";
```
Comment (anthonime) [inline swiffer-api/src/test/java/com/solambda/swiffer/api/internal/decisions/EventHandlerTypeFactoryTest.java]: should be marker1

File: swiffer-api/src/main/java/com/solambda/swiffer/api/internal/decisions/DecisionTaskContextImpl.java [hunk 2]
```diff
@@ -59,10 +66,31 @@ public VersionedName workflowType() {
...
+	}
+
 	@Override
 	public String toString() {
 		return "DecisionTaskContextImpl [domain=" + this.domain
 				+ ", workflowType=" + workflowType() + "]";
 	}
 
+    private static Predicate<HistoryEvent> isMarkerRecordedEvent(String markerName) {
+        return event -> event.getMarkerRecordedEventAttributes() != null
```
Comment (akurlykov) [inline swiffer-api/src/main/java/com/solambda/swiffer/api/internal/decisions/DecisionTaskContextImpl.java]: This method is designed to be used with markers without details. 
Replacing it with event.details() wouldn't return correct result for markers without details.

Comment (akurlykov) [inline swiffer-api/src/main/java/com/solambda/swiffer/api/internal/decisions/DecisionTaskContextImpl.java]: Also, it uses HistoryEvent, not WorkflowEvent.

Comment (anthonime) [inline swiffer-api/src/main/java/com/solambda/swiffer/api/internal/decisions/DecisionTaskContextImpl.java]: ok
done

File: swiffer-api/src/main/java/com/solambda/swiffer/api/internal/decisions/DecisionTaskContextImpl.java [hunk 3]
```diff
@@ -59,10 +66,31 @@ public VersionedName workflowType() {
...
+                        && event.getMarkerRecordedEventAttributes().getMarkerName().equals(markerName);
+    }
+
+    private <T> Function<HistoryEvent, T> deserialize(Class<T> type) {
+        return event -> dataMapper.deserialize(event.getMarkerRecordedEventAttributes().getDetails(), type);
```
Comment (anthonime) [inline swiffer-api/src/main/java/com/solambda/swiffer/api/internal/decisions/DecisionTaskContextImpl.java]: done

**Pull request merged**